src/Spaces/spectralelement.jl

```diff
@@ -283,6 +283,166 @@ end
 
 nlevels(space::SpectralElementSpace2D) = 1
 
+"""
+    SpectralElementSpace2D_sphere(topology, quadrature_style)
+
+Construct a `SpectralElementSpace2D` instance given a `topology` and `quadrature`.
+"""
+function SpectralElementSpace2D_sphere(topology, quadrature_style)
+    FT = eltype(topology.mesh)
+    CT = Geometry.LatLongPoint{FT} # Domains.coordinate_type(topology)
+    CoordType3D = Topologies.coordinate_type(topology)
+    AIdx = Geometry.coordinate_axis(CoordType3D)
+    radius = topology.mesh.domain.radius
+    nelements = Topologies.nlocalelems(topology)
+    Nq = Quadratures.degrees_of_freedom(quadrature_style)
+    # types of the partial derivative tensors
+    # ∂r∂ξ
+    Mxξ = Geometry.Axis2Tensor{
+        FT,
+        Tuple{Geometry.UVAxis, Geometry.Covariant12Axis},
+        SMatrix{2, 2, FT, 4},
+    }
+    # ∂ξ∂r
+    Mξx = Geometry.Axis2Tensor{
+        FT,
+        Tuple{Geometry.Contravariant12Axis, Geometry.UVAxis},
+        SMatrix{2, 2, FT, 4},
+    }
+    LG = Geometry.LocalGeometry{CT, FT, Mxξ, Mξx}
+    local_geometry = DataLayouts.IJFH{LG, Nq}(Array{FT}, nelements)
+    quad_points, quad_weights =
+        Quadratures.quadrature_points(FT, quadrature_style)
+    for elem in 1:nelements
+        local_geometry_slab = slab(local_geometry, elem)
+        for i in 1:Nq, j in 1:Nq
+            # compute the coordinate and partial derivative matrices for each quadrature point
+            # Guba (2014))
+            ξ = SVector(quad_points[i], quad_points[j])
+            x = Geometry.spherical_bilinear_interpolate(
+                CoordType3D.(Topologies.vertex_coordinates(topology, elem),),
+                ξ[1],
+                ξ[2],
+            )
+            xl = Geometry.LatLongPoint(x)
+            ∂x∂ξ = ForwardDiff.jacobian(ξ) do ξ
+                Geometry.components(
+                    Geometry.spherical_bilinear_interpolate(
+                        CoordType3D.(
+                            Topologies.vertex_coordinates(topology, elem),
+                        ),
+                        ξ[1],
+                        ξ[2],
+                    ),
+                )
+            end
+            if abs(xl.lat) ≈ oftype(xl.lat, 90.0)
+                # at the pole: choose u to line with x1, v to line with x2
+                ∂u∂ξ = ∂x∂ξ[SOneTo(2), :]
+            else
+                θ = xl.lat
+                λ = xl.long
+                F = @SMatrix [
+                    -sind(λ) cosd(λ) 0
+                    0 0 1/cosd(θ)
+                ]
+                ∂u∂ξ = F * ∂x∂ξ
+            end
```

Comment:
This part seems like it should be factored out so that SpectralSpace2D can work over cartesian and spherical domains, using dispatch to compute the specific metrics.